json_converter/json_mapper.py

```diff
@@ -37,13 +39,30 @@ def map(self, using={}, on=''):
                 mapping = self._apply_node_spec(item, anchor, spec)
                 if len(mapping) > 0:
                     result.append(mapping)
+                spec = copy.deepcopy(orig_spec)
         else:
             result = self._apply_node_spec(node, anchor, spec)
         return result
 
+    def map_array(self, node, using={}, on=''):
+        spec = using
+        orig_spec = copy.deepcopy(using)
+        self._check_if_readable(spec)
+        anchor = self._determine_anchor(on, spec)
+        if isinstance(node, list):
+            result = []
+            for item in node:
+                mapping = self._apply_node_spec(item, anchor, spec)
+                if len(mapping) > 0:
+                    result.append(mapping)
+                spec = copy.deepcopy(orig_spec)
+        else:
+            result = self._apply_node_spec(node, anchor, spec)
+        return result[0] if isinstance(result, list) else result
+
     @staticmethod
     def _check_if_readable(spec):
-        if not ((isinstance(spec, list) or isinstance(spec, dict)) and len(spec) > 0):
+        if not (isinstance(spec, (list, dict)) and len(spec) > 0):
```

Comment:
nice simplification here!

Reply:
Thanks, but I just checked some advices from the linter before I committed my changes ;-)